pkg/gardenlet/controller/managedseed/actuator.go

```diff
@@ -513,8 +513,9 @@ func (a *actuator) createOrUpdateSeedSecrets(ctx context.Context, spec *gardenco
 		}
 	}
 
-	// If secret reference is specified, create or update the corresponding secret
-	if spec.SecretRef != nil {
+	// If secret reference is specified and the static token kubeconfig is enabled,
+	// create or update the corresponding secret with the kubeconfig from the static token kubeconfig secret.
+	if spec.SecretRef != nil && pointer.BoolDeref(shoot.Spec.Kubernetes.EnableStaticTokenKubeconfig, false) {
```

Comment:
@rfranzke I was also wondering about the lifecycle of this Seed kubeconfig Secret that is created/updated below.
My understanding is that people rather need to put a `secretRef` to non-existing cluster and let the managedseed controller create it with the kubeconfig from the Shoot static token kubeconfig Secret.
My concern is that currently there is no deletion logic for this Secret. It has only ownerReference to the ManagedSeed and can be deleted only on ManagedSeed deletion.
I would expected at least when you set `secretRef` to nil in the gardenlet's seedConfig, the created Secret to be garbage collected somehow. Currently, it will remain in the system.
I was wondering whether we should delete this secret explicitly when the secretRef is set but the Shoot's static token kubeconfig is not enabled. At least in this way this Secret will be deleted soon or later when lock the `enableStaticTokenKubeconfig` field to `false`.

Reply:
> My understanding is that people rather need to put a secretRef to non-existing cluster and let the managedseed controller create it with the kubeconfig from the Shoot static token kubeconfig Secret

Yes, that's also my understanding.

> My concern is that currently there is no deletion logic for this Secret. It has only ownerReference to the ManagedSeed and can be deleted only on ManagedSeed deletion.
I would expected at least when you set secretRef to nil in the gardenlet's seedConfig, the created Secret to be garbage collected somehow. Currently, it will remain in the system.

I agree, I think this scenario is indeed missing in the current implementation.

> I was wondering whether we should delete this secret explicitly when the secretRef is set but the Shoot's static token kubeconfig is not enabled. At least in this way this Secret will be deleted soon or later when lock the enableStaticTokenKubeconfig field to false.

Yes, we should. Perhaps we should even forbid specifying a `secretRef` when the static kubeconfig is disabled (via the existing `ManagedSeedValidator` admission plugin).

Reply:
I don't have a proposal for the deletion logic. When `secretRef=nil` then we don't know what was the name/namespace of the Secret, hence we don't know what to delete. Maybe we should at least document this.

As the plan forward, I would rather keep the existing check as it is in the managed seed controller as a safeguard check. I also thought about validation in the ManagedSeed/ShootManagedSeed admission plugins. It make sense, let me try to add it.

Reply:
> I don't have a proposal for the deletion logic. When secretRef=nil then we don't know what was the name/namespace of the Secret, hence we don't know what to delete. Maybe we should at least document this.

Can we label the created secret to find it later for the deletion purpose?

Reply:
Yes, if we put the ManagedSeed name as label value, then it should be possible. If you don't have objections, let's tackle this in a separate issue/PR.

Reply:
The validation is added in https://github.com/gardener/gardener/pull/7546/commits/1d9a10b0397ec64b8d3e5aee347e45468816603a.
I will create a new issue for the Secret deletion.

Reply:
Alright, makes sense, thanks!